### PR TITLE
Issues/140

### PR DESCRIPTION
**Pull request by mchmielarz**

### CODE REVIEW

File: micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/IsGraphitePublishingEnabled.groovy [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+class IsGraphitePublishingEnabled implements Condition {
+
+    @Override
+    boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata) {
+        return graphitePublishingSwitchNotDefined(context) || graphitePublishingIsEnabled(context)
+    }
+
+    private boolean graphitePublishingSwitchNotDefined(ConditionContext context) {
+        !context.environment.containsProperty('graphite-publishing')
```
Comment (nurkiewicz) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/IsGraphitePublishingEnabled.groovy]: How do you set this flag? `-Dgraphite-publishing=on` in JDK?

File: micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/metrics/config/GraphitePublisherRegistrationSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    def 'should register Graphite beans when graphite-publishing is not defined'() {
+        given:
+            applicationContext.refresh()
+        expect:
+            applicationContext.getBean(GraphitePublisher) != null
+            applicationContext.getBean(Graphite) != null
+    }
+
+    def developerEnvDisabledPropertySource() {
+        return new MockPropertySource().withProperty('graphite-publishing', 'on')
```
Comment (nurkiewicz) [inline micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/metrics/config/GraphitePublisherRegistrationSpec.groovy]: Can we support more options, like `true`, `yes`, etc.? Case insensitive? Or support at least boolean, more common

File: micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/metrics/config/GraphitePublisherRegistrationSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            applicationContext.getBean(GraphitePublisher) != null
+            applicationContext.getBean(Graphite) != null
+    }
+
+    def 'should not register Graphite beans when graphite-publishing is off'() {
+        given:
+            applicationContext.environment.propertySources.addFirst(developerEnvEnabledPropertySource())
+            applicationContext.refresh()
+        expect:
+            applicationContext.getBeanNamesForType(GraphitePublisher) == []
```
Comment (nurkiewicz) [inline micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/metrics/config/GraphitePublisherRegistrationSpec.groovy]: More human readable:

    beanIsPresent(GraphitePublisher)
    beanIsAbsent(Graphite)

File: micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/IsGraphitePublishingEnabled.groovy (hunk 1 above)
Comment (mchmielarz) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/IsGraphitePublishingEnabled.groovy]: Exactly. You need to provide this flag manually if you'd like to switch off publishing (and avoid exceptions in log).

Comment (nurkiewicz) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/IsGraphitePublishingEnabled.groovy]: Cool. There are already some properties related to metrics (https://github.com/4finance/boot-microservice/blob/master/src/main/resources/application.yaml#L8-L10), may I suggest e.g. `graphite.publishing.enabled`?

File: micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/metrics/config/GraphitePublisherRegistrationSpec.groovy (hunk 2 above)
Comment (mchmielarz) [inline micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/metrics/config/GraphitePublisherRegistrationSpec.groovy]: Added.

File: micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/metrics/config/GraphitePublisherRegistrationSpec.groovy (hunk 3 above)
Comment (mchmielarz) [inline micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/metrics/config/GraphitePublisherRegistrationSpec.groovy]: Done.

File: micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/IsGraphitePublishingEnabled.groovy (hunk 1 above)
Comment (mchmielarz) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/IsGraphitePublishingEnabled.groovy]: No problem, it's a small change.

Haven't checked `application.properties` scenario. I'll write test for it.

File: micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/IsGraphitePublishingEnabled.groovy [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package com.ofg.infrastructure.metrics.config
+
+import org.springframework.context.annotation.Condition
+import org.springframework.context.annotation.ConditionContext
+import org.springframework.core.type.AnnotatedTypeMetadata
+
+class IsGraphitePublishingEnabled implements Condition {
+
+    private static final Set<String> ENABLING_VALUES = ['on', 'true', 'yes']
+    public static final String GRAPHITE_PUBLISHING = 'graphite.publishing.enabled'
```
Comment (nurkiewicz) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/IsGraphitePublishingEnabled.groovy]: Great, it should be consistent with: https://github.com/4finance/micro-infra-spring/blob/72981e8edc392ee897845fa250c353b34692c4ca/micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/MetricsConfiguration.groovy#L51

File: micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/IsGraphitePublishingEnabled.groovy [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        return graphitePublishingSwitchNotDefined(context) || graphitePublishingIsEnabled(context)
+    }
+
+    private boolean graphitePublishingSwitchNotDefined(ConditionContext context) {
+        !context.environment.containsProperty(GRAPHITE_PUBLISHING)
+    }
+
+    private boolean graphitePublishingIsEnabled(ConditionContext context) {
+        String flagValue = context.environment.getProperty(GRAPHITE_PUBLISHING).toLowerCase()
+        return ENABLING_VALUES.contains(flagValue)
```
Comment (nurkiewicz) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/IsGraphitePublishingEnabled.groovy]: Look what I found!
https://commons.apache.org/proper/commons-lang/javadocs/api-3.1/org/apache/commons/lang3/BooleanUtils.html#toBooleanObject(java.lang.String)

File: micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/IsGraphitePublishingEnabled.groovy [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        return graphitePublishingSwitchNotDefined(context) || graphitePublishingIsEnabled(context)
+    }
+
+    private boolean graphitePublishingSwitchNotDefined(ConditionContext context) {
+        !context.environment.containsProperty(GRAPHITE_PUBLISHING)
+    }
+
+    private boolean graphitePublishingIsEnabled(ConditionContext context) {
+        String graphiteEnabledValue = context.environment.getProperty(GRAPHITE_PUBLISHING).toLowerCase()
+        return ENABLING_VALUES.contains(graphiteEnabledValue)
```
Comment (nurkiewicz) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/IsGraphitePublishingEnabled.groovy]: See https://commons.apache.org/proper/commons-lang/javadocs/api-3.1/org/apache/commons/lang3/BooleanUtils.html#toBooleanObject(java.lang.String)

**Comment by nurkiewicz** [comment]
LGTM, please squash to single commit before merging (?)

**Pull request merged**